### PR TITLE
feat!: add support for settings.sql_server_audit_config setting

### PR DESCRIPTION
**Pull request by vponnam**
`settings.sql_server_audit_config` is a new feature added in the [4.28.0](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance) provider release.

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@vponnam
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vponnam
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/mssql/versions.tf [hunk 1]
```diff
@@ -20,7 +20,7 @@ terraform {
 
     google-beta = {
       source  = "hashicorp/google-beta"
-      version = ">= 4.22.0, < 5.0"
+      version = ">= 4.28.0, < 5.0"
```
Comment (g-awmalik) [inline modules/mssql/versions.tf:23]: Due to this [issue](https://github.com/hashicorp/terraform-provider-google/issues/11891)  we can't go higher than 4.24.0 right now. Let's wait on merging this until this issue is resolved.

**Comment by comment-bot-dev** [comment]
@vponnam
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/mssql/versions.tf (hunk 1 above)
Comment (g-awmalik) [reply modules/mssql/versions.tf:23]: Just realized this is the beta provider and the issue I mentioned is with GA. We should be good with this change.

**Comment by g-awmalik** [comment]
@vponnam - can you please update the upgrade to v12.0.0 doc for this change?

**Comment by comment-bot-dev** [comment]
@vponnam
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by g-awmalik — APPROVED** [review]

**Pull request merged**